src/s1reader/s1_burst_slc.py

```diff
@@ -621,6 +699,278 @@ def az_carrier_components(self, offset, position):
 
         return AzimuthCarrierComponents(kt, eta, eta_ref)
 
+    def az_fm_rate_mismatch_mitigation(self, path_dem: str,
+                                             path_scratch: str = None,
+                                             range_step=None,
+                                             az_step=None,
+                                             threshold_rdr2geo=1e-8,
+                                             numiter_rdr2geo=25,
+                                             custom_radargrid=None):
+        '''
+        Calculate azimuth FM rate mismatch mitigation
+        Based on ETAD-DLR-DD-0008, Algorithm Technical Baseline Document.
+        Available: https://sentinels.copernicus.eu/documents/247904/4629150/
+        Sentinel-1-ETAD-Algorithm-Technical-Baseline-Document.pdf
+
+        Parameters:
+        -----------
+        path_dem: str
+            Path to the DEM to calculate the actual azimuth FM rate
+        path_scratch: str
+            Path to the scratch directory to store intermediate data
+            If None, the scratch files will be saved on temporary directory generated by system
+        range_step: float
+            Range step of the correction grid in meters
+        az_step: float
+            Azimuth step of the correciton grid in seconds
+        threshold_rdr2geo: int
+            Threshold of the iteration for rdr2geo
+        numiter_rdr2geo: int
+            Maximum number of iteration for rdr2geo
+        custom_radargrid: isce3.product.RadarGridParameters
+            ISCE3 radar grid to define the correction grid.
+            If None, the full resolution radargrid of the burst will be used.
+
+        Return:
+        -------
+        _: isce3.core.LUT2d
+            azimuth FM rate mismatch rate in radar grid in seconds.
+
```

Comment:
An `Examples` section could be good here so people know it's actually pretty easy to calculate, and that many of these options aren't necessary to get some answer:

```suggestion
    Examples
    ----------
    >>> correction_grid = burst.az_fm_rate_mismatch_mitigation("my_dem.tif")
```

Reply:
Related- do we need to expose `numiter_rdr2geo`, `threshold_rdr2geo` to the user? will this be something we'll tinker with? It seems like someone may want to get corrections without knowing the implementation detail that we're using isce3's `rdr2geo` function to compute it.

Reply:
> Related- do we need to expose `numiter_rdr2geo`, `threshold_rdr2geo` to the user? will this be something we'll tinker with? It seems like someone may want to get corrections without knowing the implementation detail that we're using isce3's `rdr2geo` function to compute it.

Makes sense. I think most of the users would not need to adjust those parameters. For them, just using the default values would work most of the case. But I would rather keep this as it is because it would not harm the function, and the function would have more freedom for those who are adventurous to play with the parameters.